substrafl/experiment.py

```diff
@@ -205,6 +214,124 @@ def _get_packages_versions() -> dict:
     }
 
 
+def simulate_experiment(
+    *,
+    client: substra.Client,
+    strategy: ComputePlanBuilder,
+    train_data_nodes: List[TrainDataNodeProtocol],
+    aggregation_node: Optional[AggregationNodeProtocol] = None,
+    evaluation_strategy: Optional[EvaluationStrategy] = None,
+    num_rounds: Optional[int] = None,
+    clean_models: bool = True,
+) -> (SimuPerformancesMemory, SimuStatesMemory, SimuStatesMemory):
+    """Simulate an experiment, by computing all operation on RAM.
+    No tasks will be sent to the `Client`, which mean that this function should not be used
+    to check that your experiment will run as wanted on Substra.
+    `remote` client backend type is not supported by this function.
+
+    The intermediate states always contains the last round computed state. Set `clean_models` to `False`
+    to keep all intermediate states.
+
+    Args:
+        client (substra.Client): A substra client to interact with the Substra platform, in order to retrieve the
+            registered data. `remote` client backend type is not supported by this function.
+        strategy (Strategy): The strategy that will be executed.
+        train_data_nodes (typing.List[TrainDataNodeProtocol]): List of the nodes where training on data
+            occurs.
+        aggregation_node (typing.Optional[AggregationNodeProtocol]): For centralized strategy, the aggregation
+            node, where all the shared tasks occurs else None.
+        evaluation_strategy (EvaluationStrategy, Optional): If None performance will not be measured at all.
```

Comment:
Nipick but it would be better to have some uniformity between parameters. Either use the `<type>, Optional` syntax for both aggregation_node and evaluation_strategy, or the `typing.Optional` syntax. 
I also think that the `typing.` precision before all types is superfluous, as it is clear from context that these are types, and that it is just needlessly verbose.